### ISSUE TITLE
zsh cd completions duplicate with group-name enabled

### ISSUE THREAD

**Issue by grimm26** [ajeetdsouza/zoxide#491]
I stripped down my completions setup in zshrc to be just:
```
autoload -Uz compinit && compinit -C
eval "$(zoxide init -zsh)"
```
Now, I add a format and group-name setting and the strange behavior starts.
When using `cd` tab completion, I get expected suggestions. When using `z`, I get a duplicate set of suggestions.
![Screenshot from 2022-12-05 08-30-00](https://user-images.githubusercontent.com/127401/205664059-e3bd051c-2a2c-4c99-90cc-6443559d691d.png)
![Screenshot from 2022-12-05 08-29-46](https://user-images.githubusercontent.com/127401/205664083-1b9e3281-2d91-44aa-b86a-ac5225896499.png)

**Comment by ericbn** [comment]
For reference, this is how to reproduce:
```shell
$ cd $(mktemp -d)
$ touch .zshrc
$ ZDOTDIR=${PWD} HOME=${PWD} exec zsh
$ autoload -Uz compinit && compinit -C
$ eval "$(zoxide init zsh)"
$ zstyle ':completion:*' format '%F{yellow}-- %d --%f'
$ zstyle ':completion:*' group-name ''
$ mkdir test{1,2}
$ z test<TAB>
-- file --
test1/  test2/
-- file --
test1/  test2/
```

@grimm26, I noticed it works as expected if you do `eval "$(zoxide init zsh)"` before `autoload -Uz compinit && compinit -C`.

**Comment by grimm26** [comment]
@ericbn You are correct! I have no idea how/why but it works. Maybe just a documentation change is needed?

**Comment by timkgh** [comment]
> I noticed it works as expected if you do eval "$(zoxide init zsh)" before autoload -Uz compinit && compinit -C

Not easy to do if you use [prezto](https://github.com/sorin-ionescu/prezto) which does the autoload in its modules. [ohmyzsh](https://github.com/ohmyzsh/ohmyzsh) _may_ have the same issue too.

**Comment by timkgh** [comment]
FWIW, `zi` doesn't have the duplicate behavior.

**Comment by Subjective** [comment]
I'm also experiencing this issue, but calling `eval "$(zoxide init zsh)"` before `autoload -Uz compinit && compinit -C` seems to completely break completions for me.

**Comment by Squirreljetpack** [comment]
I found a solution haha
After compinit. compdef __zoxide_z=vim; compdef __zoxide_zi=vim;

**Comment by Kranzes** [comment]
> I found a solution haha. I'm using zi so zoxide is run before compinit, but I still got the duplicate. Turns out the __z commands default to a certain completion if no compdef is provided. To override this, add this: compdef __zoxide_z=vim; compdef __zoxide_zi=vim;.

That's not a solution, this makes it so the completions for zoxide will be the completions of vim.

**Comment by Kranzes** [comment]
> I'm also experiencing this issue, but calling `eval "$(zoxide init zsh)"` before `autoload -Uz compinit && compinit -C` seems to completely break completions for me.

Same

**Comment by mikegreiling** [comment]
this bug is quite annoying, I'm surprised more people have not been running into it

**Comment by Kranzes** [comment]
This solved it for me #787

**Comment by urob** [comment]
> This solved it for me #787

Seems to be *exactly* the fix that I have posted [above](https://github.com/ajeetdsouza/zoxide/issues/491#issuecomment-1973831206). Glad someone picked it up and submitted a PR.

**Comment by grimm26** [comment]
I do notice that using `_cd` vs `_path_file` gives slightly different result in that it is titled with local when using `_cd`. :shrug:

**Comment by urob** [comment]
> I do notice that using `_cd` vs `_path_file` gives slightly different result in that it is titled with local when using `_cd`. :shrug:

Yes, nevermind my previous comment. Not sure where my head was. I remember considering just applying the builtin autocompletion (like the PR does) and then discarding it for `_path_file`. I don't quite recall my reasoning atm, will look into it later.

**Comment by Kemboiray** [comment]
> eval "$(zoxide init --cmd cd zsh | sed 's/_files/_cd/g')"

This works for me. Thanks.

**Comment by ajeetdsouza** [comment]
@Kemboiray is your issue not solved on the latest version of zoxide?

**Comment by Kemboiray** [comment]
@ajeetdsouza I’m not on the latest, as I installed with `apt`, which provides `0.9.3`.

**Comment by ajeetdsouza** [comment]
Closing since this is fixed in the latest version.
